go-controller/pkg/ovn/loadbalancer/lb_cache.go

```diff
@@ -93,6 +100,50 @@ func (c *LBCache) removeVips(toRemove []DeleteVIPEntry) {
 	}
 }
 
+// RemoveSwitches removes the provided toRemoveSwitches which is a list
+// of switches from all the lb's.Switches in the LBCache.
+func (c *LBCache) RemoveSwitches(toRemoveSwitches []string) {
+	c.Lock()
+	defer c.Unlock()
+	for _, sw := range toRemoveSwitches {
+		switches, err := findTableLBs("logical_switch", sw)
```

Comment:
Rather than searching through ovsdb, which might be inconsistent with the cache, just walk through the cache and manually remove all switches from the cache entries.

Reply:
agreed, no need for extra db calls.